modules/urlsigning-service-impl/src/main/java/org/opencastproject/security/urlsigning/provider/impl/AbstractUrlSigningProvider.java

```diff
@@ -130,62 +146,69 @@ public void updated(Dictionary<String, ?> properties) throws ConfigurationExcept
     }
 
     // Collect configuration in a new map so we don't partially override the old one in case of error
-    Map<String, KeyEntry> keys = new HashMap<>();
+    TreeMap<String, Key> urls = new TreeMap<>();
+
+    // Temporary list of key entries to simplify building up the keys
+    Map<String, Key> keys = new HashMap<>();
 
     Enumeration<String> propertyKeys = properties.keys();
     while (propertyKeys.hasMoreElements()) {
       String propertyKey = propertyKeys.nextElement();
 
-      String keyEntryProperty = StringUtils.removeStart(propertyKey, KEY_ENTRY_PREFIX + ".");
-      if (keyEntryProperty == propertyKey) continue;
+      if (!propertyKey.startsWith(KEY_PROPERTY_PREFIX + ".")) {
```

Comment:
No, this change is not related to performance but rather about code readability:

```
       String keyEntryProperty = StringUtils.removeStart(propertyKey, KEY_ENTRY_PREFIX + ".");
       if (keyEntryProperty == propertyKey) continue;
      // This configuration property key starts with KEY_ENTRY_PREFIX + "." so we have to handle it
```

just seems a crazy way to say

```
  if (!propertyKey.startsWith(KEY_PROPERTY_PREFIX + ".")) {
      // This configuration property key starts withKEY_ENTRY_PREFIX + "."so we have to handle it
```

to me.

Reply:
I've changed it to

```
  if (propertyKey.startsWith(KEY_PROPERTY_PREFIX + ".")) {
      // This configuration property key starts with KEY_ENTRY_PREFIX + "."so we have to handle it

```
so that the code matches the semantics.

Reply:
Okay, I take your first point, but (and we had this discussion already) I really think that

```
if (condition) {
  // many many lines of code
  // that handle the thing
  // and introduce multiple further levels of indentation
  // [...]
        }
      }
    }
  }
}
// Where does `if (condition)` end?! WHERE AM I?!! O_O
```

**much** less readable than early bailouts. :thinking:

Again, not gonna insist here, but weren't you the one who was against changing things not directly related to the PR? :wink: